### ISSUE TITLE
`skaffold init --analyze --generate-manifests` does not emit images

### ISSUE THREAD

**Issue by j-windsor** [GoogleContainerTools/skaffold#6249]
### Expected behavior
Since I have expressed my interest to generate the manifests with the `--generate-manifests` flag, I should see the `images` object in the output with the expected image names after the manifests are output, so I can properly specify the `--artifact` flag on my next call to `skaffold init --artifact=object --generate-manifests`.

```
➜  hello-world-55 skaffold init --analyze --generate-manifests
{"builders":[{"name":"Docker","payload":{"path":"Dockerfile"}},{"name":"Buildpacks","payload":{"path":"package.json","builder":"gcr.io/buildpacks/builder:v1"}}],"images":[{"name":"myapp","foundMatch":false}]}
```

### Actual behavior
```
➜  hello-world-55 skaffold init --analyze --generate-manifests
{"builders":[{"name":"Docker","payload":{"path":"Dockerfile"}},{"name":"Buildpacks","payload":{"path":"package.json","builder":"gcr.io/buildpacks/builder:v1"}}]}
```
### Information

- Skaffold version: v1.27.0
- Operating system: MacOS Big Sur 11.4
- Installed via: Google Cloud SDK
- Contents of skaffold.yaml: n/a

### Steps to reproduce the behavior

1. Clone https://github.com/GoogleCloudPlatform/cloud-code-samples/tree/master/nodejs/nodejs-hello-world
2. `skaffold init --analyze --generate-manifests`

**Comment by MarlonGamez** [comment]
With this issue, and https://github.com/GoogleContainerTools/skaffold/issues/6251, it's clear that we have a few issue with mixing flags when it comes to init. I'll try to take a look and see how we can untangle these features.

**Comment by MarlonGamez** [comment]
Hi @j-windsor , I looked into this a bit and I realized that this only happens when the kubernetes manifests are removed from the project. I originally thought this was happening even when there were manifests existing within the project. 

The reason this happens is that since there are no kubernetes manifests to pull image names from, we cannot provide any images in the output from `skaffold init --analyze`.

I think the ideal workflow here is to run `skaffold init --analyze`, give the information to the user through CC UI, and allow them to fill out whatever information necessary for us to run `skaffold init -a ...` for their project. The API for `skaffold init -a ...` should be expanded to support taking some extra information to allow for manifest generation.
